src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -102,10 +102,74 @@ struct ExternalFlashFactoryData
     uint8_t mFactoryDataBuffer[PM_FACTORY_DATA_SIZE];
 };
 
+class FactoryDataProviderProxy : public chip::Credentials::DeviceAttestationCredentialsProvider,
```

Comment:
nit: technically it's not a proxy pattern. It's more `Base` or `Interface`.

Reply:
I see, proxy should actually do some translation between interfaces or something similar. I will change that to *Base.